webhook/server/server.go

```diff
@@ -0,0 +1,286 @@
+	if err := m.client.Get(ctx, client.ObjectKey{Name: oaName, Namespace: m.namespace}, &oa); k8serrors.IsNotFound(err) {
+		return admission.Errored(http.StatusBadRequest, fmt.Errorf(
+			"namespace '%s' is assigned to DynaKube instance '%s' but doesn't exist", req.Namespace, oaName))
+	} else if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if pod.Annotations == nil {
+		pod.Annotations = map[string]string{}
+	}
+
+	if pod.Annotations[dtwebhook.AnnotationInjected] == "true" {
+		return admission.Patched("")
+	}
+	pod.Annotations[dtwebhook.AnnotationInjected] = "true"
+
+	technologies := url.QueryEscape(utils.GetField(pod.Annotations, dtwebhook.AnnotationTechnologies, "all"))
+	installPath := utils.GetField(pod.Annotations, dtwebhook.AnnotationInstallPath, dtwebhook.DefaultInstallPath)
+	installerURL := utils.GetField(pod.Annotations, dtwebhook.AnnotationInstallerUrl, "")
+	failurePolicy := utils.GetField(pod.Annotations, dtwebhook.AnnotationFailurePolicy, "silent")
+	image := m.image
+
+	pod.Spec.Volumes = append(pod.Spec.Volumes,
+		corev1.Volume{
+			Name: "init",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		corev1.Volume{
+			Name: "oneagent",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+		corev1.Volume{
+			Name: "oneagent-config",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: dtwebhook.SecretConfigName,
+				},
+			},
+		})
+
+	var sc *corev1.SecurityContext
+	if pod.Spec.Containers[0].SecurityContext != nil {
+		sc = pod.Spec.Containers[0].SecurityContext.DeepCopy()
+	}
+
+	fieldEnvVar := func(key string) *corev1.EnvVarSource {
+		return &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: key}}
+	}
+
+	basePodName := pod.GenerateName
+	if basePodName == "" {
+		basePodName = pod.Name
+	}
+
+	// Only include up to the last dash character, exclusive.
+	if p := strings.LastIndex(basePodName, "-"); p != -1 {
+		basePodName = basePodName[:p]
+	}
+
+	ic := corev1.Container{
+		Name:            "install-oneagent",
+		Image:           image,
+		ImagePullPolicy: corev1.PullAlways,
+		Command:         []string{"/usr/bin/env"},
+		Args:            []string{"bash", "/mnt/config/init.sh"},
+		Env: []corev1.EnvVar{
+			{Name: "TECHNOLOGIES", Value: technologies},
+			{Name: "INSTALLPATH", Value: installPath},
+			{Name: "INSTALLER_URL", Value: installerURL},
+			{Name: "FAILURE_POLICY", Value: failurePolicy},
+			{Name: "CONTAINERS_COUNT", Value: strconv.Itoa(len(pod.Spec.Containers))},
+			{Name: "K8S_PODNAME", ValueFrom: fieldEnvVar("metadata.name")},
+			{Name: "K8S_PODUID", ValueFrom: fieldEnvVar("metadata.uid")},
+			{Name: "K8S_BASEPODNAME", Value: basePodName},
+			{Name: "K8S_NAMESPACE", ValueFrom: fieldEnvVar("metadata.namespace")},
+			{Name: "K8S_NODE_NAME", ValueFrom: fieldEnvVar("spec.nodeName")},
```

Comment:
the removal of `USE_IMMUTABLE_IMAGE` environment variable breaks the script generated by `namespace/namespace_controller.go`

Reply:
Fixed.